### PR TITLE
Fix(tests) Improve panel pytest

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description

As part of the brown bad seminar I refactored a test for the panel function. This PR adds the new test and removes the old one
### Added

- New test for the `panel` function

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/cli/workflow/mip/test_cli_mip_dna_panel.py [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+
+import mock
+from click.testing import CliRunner
 
 from cg.cli.workflow.mip_dna.base import panel
+from cg.meta.workflow.mip import MipAnalysisAPI
+from cg.models.cg_config import CGConfig
+from tests.conftest import create_process_response
+
+LOG = logging.getLogger(__name__)
```
Comment (islean) [inline tests/cli/workflow/mip/test_cli_mip_dna_panel.py:13]: Is this used?

**Review by islean — APPROVED** [review]
Great!

File: tests/cli/workflow/mip/test_cli_mip_dna_panel.py [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
-    # WHEN we run a case in dry run mode
-    result = cli_runner.invoke(panel, [case_id], obj=mip_dna_context)
+    # THEN the file should exist
+    assert panel_file.exists()
 
-    # THEN the command should be printed
-    assert result.exit_code == 0
+    # THEN the file should contain the output from scout
+    with open(panel_file, "r") as file_handle:
+        assert file_handle.read() == scout_panel_output
```
Comment (ChrOertlin) [inline tests/cli/workflow/mip/test_cli_mip_dna_panel.py:38]: don't we have a file reader for this?

File: tests/conftest.py [hunk 3]
```diff
@@ -3259,3 +3263,14 @@ def flow_cell_encryption_api(
     )
     flow_cell_encryption_api.slurm_api.set_dry_run(dry_run=True)
     return flow_cell_encryption_api
+
+
+def create_process_response(
+    return_code: int = 0, args: str = "", std_out: str = "", std_err: str = ""
```
Comment (ChrOertlin) [inline tests/conftest.py:3269]: could use a docstring so the use case is clear

**Review by ChrOertlin — APPROVED** [review]
Looks good!

File: tests/cli/workflow/mip/test_cli_mip_dna_panel.py (hunk 1 above)
Comment (Vince-janv) [reply tests/cli/workflow/mip/test_cli_mip_dna_panel.py:13]: Nope, removed 👍

File: tests/cli/workflow/mip/test_cli_mip_dna_panel.py (hunk 2 above)
Comment (Vince-janv) [reply tests/cli/workflow/mip/test_cli_mip_dna_panel.py:38]: Not sure if we have one for just reading a whole file

File: tests/conftest.py (hunk 3 above)
Comment (Vince-janv) [reply tests/conftest.py:3269]: Added! 👍

File: tests/cli/workflow/mip/test_cli_mip_dna_panel.py (hunk 2 above)
Comment (Vince-janv) [reply tests/cli/workflow/mip/test_cli_mip_dna_panel.py:38]: Plot-twist! There was. Used it here

**Pull request merged**